packages/neon_framework/lib/src/utils/image_utils.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:xml/xml.dart';
+import 'package:xml/xpath.dart';
+
+/// Utilities for working with images.
+class ImageUtils {
+  ImageUtils._();
+
+  /// Rewrites percent dimensions in SVGs to be absolute.
+  ///
+  /// This is required because the `flutter_svg` package can not handle percent dimensions.
+  static String rewriteSvgDimensions(String data) {
+    final document = XmlDocument.parse(data);
+
+    final viewBoxAttribute = [
+      ...document.xpath('/svg/@viewBox'),
+      ...document.xpath('/svg/@viewbox'),
+    ].first as XmlAttribute;
+    final viewBoxValues = viewBoxAttribute.value.split(' ');
```

Comment:
Why add it to a list only to discard it a bit later? If we find a match in the we'd also save a second lookup.
```suggestion
    final viewBoxAttribute = document.xpath('/svg/@viewBox').singleOrNull ?? document.xpath('/svg/@viewbox').singleOrNull;
    final viewBoxValues = viewBoxAttribute!.value!.split(' ');
```

But I'm not sure this is worth it and your approach might be somewhat more readable.

Reply:
Yeah I originally did it for readability, but right now it is readable in this way too.